gdx/src/com/badlogic/gdx/graphics/g3d/utils/MeshBuilder.java

```diff
@@ -557,8 +560,18 @@ public short vertex (Vector3 pos, Vector3 nor, Color col, Vector2 uv) {
 			vertex[colOffset + 2] = col.b;
 			if (colSize > 3) vertex[colOffset + 3] = col.a;
 		} else if (cpOffset > 0) {
-			if (col == null) col = Color.WHITE;
-			vertex[cpOffset] = col.toFloatBits(); // FIXME cache packed color?
+			if (col == null) {
+				vertex[cpOffset] = Color.WHITE_FLOAT_BITS;
+			} else {
+				final int colIntBits = col.toIntBits();

```

Comment:
Maybe you could simplify this if/else block a bit : 
```java
if(colIntBits != lastColIntBits) {
    lastColIntBits = colIntBits;
    cacheFloatBits = NumberUtils.intToFloatColor(colIntBits);
}
vertex[cpOffset] = cacheFloatBits;
```

Reply:
Yaa, I noticed it possible after I created pull requests, I just so tired after hours of figure it out most optimize ways to cache packed color after I saw FIXME comment. Really, it looks like they left it for me! after you point it out, I'll simplify the code. 👍